packages/demo/backend/src/services/wallet.ts

```diff
@@ -13,64 +10,93 @@ import {
+  cursor?: string
+}
 
 export async function createWallet(): Promise<{
   privyAddress: string
   smartWalletAddress: string
 }> {
   const verbs = getVerbs()
-  const wallet = await verbs.wallet.createWalletWithHostedSigner()
-  const smartWalletAddress = await wallet.getAddress()
+  const privyClient = getPrivyClient()
+  const privyWallet = await privyClient.walletApi.createWallet({
+    chainType: 'ethereum',
+  })
+  const verbsPrivyWallet = await verbs.wallet.hostedWalletToVerbsWallet({
+    walletId: privyWallet.id,
+    address: privyWallet.address,
+  })
+  const wallet = await verbs.wallet.createSmartWallet({
+    owners: [verbsPrivyWallet.address],
+    signer: verbsPrivyWallet.signer,
+  })
+  const smartWalletAddress = wallet.address
   return {
     privyAddress: wallet.signer.address,
     smartWalletAddress,
   }
 }
 
-export async function getWallet(userId: string): Promise<{
-  wallet: SmartWallet
-}> {
+export async function getWallet(userId: string): Promise<SmartWallet | null> {
   const verbs = getVerbs()
-  const wallet = await verbs.wallet.getSmartWalletWithHostedSigner({
-    walletId: userId,
+  const privyClient = getPrivyClient()
+  const privyWallet = await privyClient.walletApi
+    .getWallet({
+      id: userId,
+    })
+    .catch(() => null)
+  if (!privyWallet) {
+    return privyWallet
+  }
+  const verbsPrivyWallet = await verbs.wallet.hostedWalletToVerbsWallet({
+    walletId: privyWallet.id,
+    address: privyWallet.address,
   })
-  return { wallet }
+  const wallet = await verbs.wallet.getSmartWallet({
+    signer: verbsPrivyWallet.signer,
+    deploymentOwners: [getAddress(privyWallet.address)],
+  })
+  return wallet
 }
 
 export async function getAllWallets(
   options?: GetAllWalletsOptions,
 ): Promise<Array<{ wallet: SmartWallet; id: string }>> {
   try {
     const verbs = getVerbs()
-    const privyWallets = await (
-      verbs.wallet.hostedWalletProvider as PrivyHostedWalletProvider
-    ).getAllWallets(options)
+    const privyClient = getPrivyClient()
+    const response = await privyClient.walletApi.getWallets(options)
     return Promise.all(
-      privyWallets.map(async (privyWallet) => {
-        const walletAddress =
-          await verbs.wallet.smartWalletProvider.getWalletAddress({
-            owners: [privyWallet.address],
-          })
-        const account = await privyWallet.account()
-        const wallet = (
-          verbs.wallet.smartWalletProvider as SmartWalletProvider
-        ).getWallet({
-          walletAddress,
-          signer: account,
-          ownerIndex: 0,
+      response.data.map(async (privyWallet) => {
+        const verbsPrivyWallet = await verbs.wallet.hostedWalletToVerbsWallet({
+          walletId: privyWallet.id,
+          address: privyWallet.address,
+        })
+        const wallet = await verbs.wallet.getSmartWallet({
+          signer: verbsPrivyWallet.signer,
+          deploymentOwners: [getAddress(privyWallet.address)],
```

Comment:
🤔 Instead of the dev calling:
```
const privyWallet = verbs.wallet.hostedWalletToVerbsWallet(...)
const wallet = await verbs.wallet.getSmartWallet({...})
```
could `hostedWalletToVerbsWallet` return the smartWallet as well? like
```
const privyWallet = verbs.wallet.hostedWalletToVerbsWallet(...)
const wallet = privyWallet.smartWallet()
```

Separate from this specific case, but a general thought I've had, I'm worried about devs juggling context between or even confusing HostedWallets, PrivyWallets, VerbsWallets, SmartWallets. Our goal should be to make it *very* difficult for a dev to accidentally put money in the user's wrong sub-wallet, though I'm not sure how to achieve that goal.

Reply:
I'm open to adding that if there is a preference from users to do that and this ends up being a painpoint that is easy for them to get wrong. 

However, I actually think that converting a hostedWallet automatically to a smart wallet is more likely to result in money being accidentally put in the wrong sub-wallet. By separating this into two separate functions and not trying to combine so many side effects into one action, it forces the dev to be explicit about the configuration of their smart wallet. Also, if we were to combine both into one function we'd have to overload the `hostedWalletToVerbsWallet` with the arguments that `getSmartWallet` uses and the interface becomes a bit confusing.

Reply:
make sense to me, lets keep it